File: solutions/ruby/resistor-color/1/resistor_color.rb
# ResistorColor class
class ResistorColor
  COLORS = %w[
      black
      brown
      red
      orange
      yellow
      green
      blue
      violet
      grey
      white
  ]

  class << self
    def color_code(color)
      COLORS.index(color)
    end
  end
end

File: solutions/go/resistor-color/1/resistor_color.go
package resistorcolor

var colors []string

func init() {
	colors = []string{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}
}

// Colors should return the list of all colors.
func Colors() []string { return colors }

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	for i, c := range colors {
		if c == color {
			return i
		}
	}
	return -1
}
